lib/ansible/module_utils/powershell/Ansible.ModuleUtils.WebRequest.psm1

```diff
@@ -467,58 +465,9 @@ Function Invoke-WithWebRequest {
     }
 }
 
-Function Merge-WebRequestSpec {
-    <#
-    .SYNOPSIS
-    Merges a modules spec definition with extra options supplied by this module_util. Options from the module take
-    priority over the module util spec.
-
-    .PARAMETER ModuleSpec
-    The root $spec of a module option definition to merge with.
-
-    .EXAMPLE
-    $spec = @{
-        options = @{
-            name = @{ type = "str" }
-        }
-        supports_check_mode = $true
-    }
-    $spec = Merge-WebRequestSpec -ModuleSpec $spec
-    #>
-    [CmdletBinding()]
-    param (
-        [Parameter(Mandatory=$true)]
-        [System.Collections.IDictionary]
-        $ModuleSpec,
-
-        [System.Collections.IDictionary]
-        $SpecToMerge = @{ options = $ansible_web_request_options }
-    )
-
-    foreach ($option_kvp in $SpecToMerge.GetEnumerator()) {
-        $k = $option_kvp.Key
-        $v = $option_kvp.Value
-
-        if ($ModuleSpec.Contains($k)) {
-            if ($v -is [System.Collections.IDictionary]) {
-                $ModuleSpec[$k] = Merge-WebRequestSpec -ModuleSpec $ModuleSpec[$k] -SpecToMerge $v
-            } elseif ($v -is [Array] -or $v -is [System.Collections.IList]) {
-                $sourceList = [System.Collections.Generic.List[Object]]$ModuleSpec[$k]
-                foreach ($entry in $v) {
-                    $sourceList.Add($entry)
-                }
-
-                $ModuleSpec[$k] = $sourceList
-            }
-        } else {
-            $ModuleSpec[$k] = $v
-        }
-    }
-
-    $ModuleSpec
-}
-
 # See lib/ansible/plugins/doc_fragments/url_windows.py
+# Kept here for backwards compat as this variable was added in Ansible 2.9. Ultimately this util should be removed
+# once the deprecation period has been added.
```

Comment:
Are we keeping this just in case something outside of ansible[-collections] was using it?

Reply:
Yep, technically it was added in 2.9 and without a deprecation period we can't remove it. Once this PR is in I'm hoping to create a copy of this util in ansible.windows and stop it relying on what's in ansible/ansible. The end goal is to move away from module utils being in ansible/ansible unless they are critical for all module execution, which this one is not.